lib/elixir/lib/gen_event.ex

```diff
@@ -783,7 +784,7 @@ defmodule GenEvent do
   end
 
   @doc false
-  def system_code_change([name, handlers, hib], module, old_vsn, extra) do
+  def system_code_change([name, handlers, hib], module, old_vsn, extra) when is_atom(module) do
```

Comment:
This callback is called by `:sys` and we should trust it not to pass a bad module. The place to check that it is valid module would be in `:sys.change_code`.